### DISCUSSION TITLE
Sveltekit Supabase array not contains

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#8796] [Questions] [answered]
Using Sveltekit the contains attribute is quite powerful for array type columns. However Im struggling to find a solution for not contains.

Tried for example:

.not('ARRAYCOL','contains','banana')

Is it not doable?..

**Comment by GaryAustin1** [comment, marked as answer]
https://supabase.com/docs/reference/javascript/not#notes
You need to use raw PostgREST format in .not().  So 'cs' instead of contains.  
Contains also requires an array for the comparison even if a single value and the example shows using a string for the array (not sure if a javascript array will work).  `'{"banana"}'` is the Postgres string format for an array with one item.

**Reply by ghost** [reply]
Thank you.

.not('ARRAYCOL','cs','{"banana"}')

works. As does

.not('ARRAYCOL','cs','{"banana","orange"}')

For and.

Now trying to figure out how to do a not contains "banana" or "orange", for now doing

.not('ARRAYCOL','cs','{"banana"}')
.not('ARRAYCOL','cs','{"orange"}')
 
for that.

**Reply by GaryAustin1** [reply]
That is the most straight forward way...  not (a or b) = not(a) and not(b)
